sklearn/tests/test_public_functions.py

```diff
@@ -0,0 +1,77 @@
+from importlib import import_module
+from inspect import signature
+
+import pytest
+
+from sklearn.utils._param_validation import generate_invalid_param_val
+from sklearn.utils._param_validation import generate_valid_param
+from sklearn.utils._param_validation import make_constraint
+from sklearn.cluster import kmeans_plusplus
+
+
+PARAM_VALIDATION_FUNCTION_LIST = [kmeans_plusplus]
+
+
+@pytest.mark.parametrize("func", PARAM_VALIDATION_FUNCTION_LIST)
+def test_function_param_validation(func):
+    """Check that an informative error is raised when the value of a parameter does not
+    have an appropriate type or value.
+
+    the parameter constraints for the function must be named
+    _parameter_constraints_of_<func_name> and be located in the same module.
+    """
+    func_sig = signature(func)
+    func_params = [
+        p.name
+        for p in func_sig.parameters.values()
+        if p.kind not in (p.VAR_POSITIONAL, p.VAR_KEYWORD)
```

Comment:
It could be nice to document that we don't check `*args` and `**kwargs` for constraints.

Reply:
done in `validate_params`